File: frontend/src/lib/api/revenue-level-tier.ts
import apiClient from './client';

export interface RevenueLevelTier {
  id: string;
  level_number: number;
  level_name: string;
  min_revenue: number;
  max_revenue: number | null;
  display_order: number;
  color_code: string | null;
  description: string | null;
  created_at: string;
  updated_at: string;
}

export interface RevenueLevelTierCreate {
  level_number: number;
  level_name: string;
  min_revenue: number;
  max_revenue?: number | null;
  display_order?: number;
  color_code?: string | null;
  description?: string | null;
}

export interface RevenueLevelTierUpdate {
  level_name?: string;
  min_revenue?: number;
  max_revenue?: number | null;
  display_order?: number;
  color_code?: string | null;
  description?: string | null;
}

export const revenueLevelTierApi = {
  list: async () => {
    const response = await apiClient.get('/revenue-level-tiers');
    return response.data as RevenueLevelTier[];
  },

  getById: async (id: string) => {
    const response = await apiClient.get(`/revenue-level-tiers/${id}`);
    return response.data as RevenueLevelTier;
  },

  create: async (data: RevenueLevelTierCreate) => {
    const response = await apiClient.post('/revenue-level-tiers', data);
    return response.data as RevenueLevelTier;
  },

  update: async (id: string, data: RevenueLevelTierUpdate) => {
    const response = await apiClient.put(`/revenue-level-tiers/${id}`, data);
    return response.data as RevenueLevelTier;
  },

  delete: async (id: string) => {
    const response = await apiClient.delete(`/revenue-level-tiers/${id}`);
    return response.data;
  },

  getTierForRevenue: async (revenue: number) => {
    const response = await apiClient.post('/revenue-level-tiers/match', { revenue });
    return response.data as RevenueLevelTier;
  },
};

File: backend/internal/domain/models/revenue_level_tier.go
package models

import (
	"time"

	"github.com/google/uuid"
)

// RevenueLevelTier represents a revenue level tier configuration
type RevenueLevelTier struct {
	ID          uuid.UUID  `json:"id" db:"id"`
	LevelNumber int        `json:"level_number" db:"level_number"`
	LevelName   string     `json:"level_name" db:"level_name"`
	MinRevenue  float64    `json:"min_revenue" db:"min_revenue"`
	MaxRevenue  *float64   `json:"max_revenue,omitempty" db:"max_revenue"`
	DisplayOrder int       `json:"display_order" db:"display_order"`
	ColorCode   *string    `json:"color_code,omitempty" db:"color_code"`
	Description *string    `json:"description,omitempty" db:"description"`
	CreatedAt   time.Time  `json:"created_at" db:"created_at"`
	UpdatedAt   time.Time  `json:"updated_at" db:"updated_at"`
}

// RevenueLevelTierCreate represents data for creating a new revenue level tier
type RevenueLevelTierCreate struct {
	LevelNumber int     `json:"level_number" binding:"required,min=1,max=10"`
	LevelName   string  `json:"level_name" binding:"required,max=50"`
	MinRevenue  float64 `json:"min_revenue" binding:"required,min=0"`
	MaxRevenue  *float64 `json:"max_revenue,omitempty"`
	DisplayOrder int     `json:"display_order"`
	ColorCode   *string  `json:"color_code,omitempty"`
	Description *string  `json:"description,omitempty"`
}

// RevenueLevelTierUpdate represents data for updating a revenue level tier
type RevenueLevelTierUpdate struct {
	LevelName   *string  `json:"level_name,omitempty"`
	MinRevenue  *float64 `json:"min_revenue,omitempty"`
	MaxRevenue  *float64 `json:"max_revenue,omitempty"`
	DisplayOrder *int    `json:"display_order,omitempty"`
	ColorCode   *string  `json:"color_code,omitempty"`
	Description *string  `json:"description,omitempty"`
}
